code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -49,15 +76,70 @@
 /obj/item/reagent_containers/food/drinks/cans/afterattack(obj/target, mob/user, proximity)
 	if(!proximity)
 		return
-	if(istype(target, /obj/structure/reagent_dispensers) && (canopened == 0))
+	if(istype(target, /obj/structure/reagent_dispensers) && !canopened)
 		to_chat(user, "<span class='notice'>You need to open the drink!</span>")
 		return
-	else if(target.is_open_container() && (canopened == 0))
+	else if(target.is_open_container() && !canopened)
 		to_chat(user, "<span class='notice'>You need to open the drink!</span>")
 		return
 	else
 		return ..(target, user, proximity)
 
+/obj/item/reagent_containers/food/drinks/cans/after_throw(mob/user)
```

Comment:
This one needs to call it's parent at one point since that handles quite some logic. 
The param here is also incorrect:
![image](https://user-images.githubusercontent.com/15887760/96304779-0205a900-0ffd-11eb-8d89-f1a1b1823d0d.png)
is where the base is defined and called

Reply:
I switched to using `throw_impact()` hopefully for the better